### PR TITLE
SELECT name, change a switch to if/else, make a JSON struct more explicit

### PR DESCRIPTION
**Pull request by ijt**
The original PR chooses a set of repositories to index on sourcegraph.com.

Test plan: The changes are covered by existing tests.

### CODE REVIEW

File: cmd/frontend/internal/httpapi/internal.go [hunk 1]
```diff
@@ -209,18 +208,20 @@ func serveReposList(w http.ResponseWriter, r *http.Request) error {
...
-	// BACKCOMPAT: Add a "URI" field because zoekt-sourcegraph-indexserver expects one to exist
-	// (with the repository name). This is a legacy of the rename from "repo URI" to "repo name".
+	// BACKCOMPAT: Add a Name field that serializes to `URI` because
+	// zoekt-sourcegraph-indexserver expects one to exist (with the
+	// repository name). This is a legacy of the rename from "repo URI" to
+	// "repo name".
 	type repoWithBackcompatURIField struct {
-		URI string
-		*types.Repo
+		Name string `json:"URI"`
```
Comment (keegancsmith) [inline cmd/frontend/internal/httpapi/internal.go]: This needs to be URI. Zoekt expects it to be called URI, even though it is the name. That is why this 
hack exists. So just revert this part of the change, but keep the `SELECT name` in the db package.

Comment (ijt) [reply cmd/frontend/internal/httpapi/internal.go]: Isn't the change I made here effectively what you were asking for in https://github.com/sourcegraph/sourcegraph/pull/4959#discussion_r307237993?

I guess you changed your mind about this. What happened?

Comment (ijt) [reply cmd/frontend/internal/httpapi/internal.go]: I think your earlier suggestion makes more sense so I'm leaving it alone for now.

**Comment by ijt** [comment]
> Thanks. Approving to unblock, but important inline feedback.

Thanks for approving. What is the importance of the feedback? It seems to have no observable effects and contradicts the advice you gave previously, unless I misunderstood.

> BTW did you look into moving this special casing for (sourcegraph.com + index) into the DB layer?

I haven't yet and I'm still not sure how that would be better. As far as I can tell, the way it's structured now is fine. Okay, maybe we really should do that meeting...

File: cmd/frontend/internal/httpapi/internal.go (hunk 1 above)
Comment (keegancsmith) [reply cmd/frontend/internal/httpapi/internal.go]: Oh sorry I totally missed the `json:"URI"` part. This looks good :)

**Comment by keegancsmith** [comment]
Sorry when I reviewed I missed the `json:URI` bit. So this looks good to me. Could you give this PR a better title? I know it has a few different changes in it, but maybe the title can be the most important change which is to `SELECT name`.

**Comment by ijt** [comment]
> Sorry when I reviewed I missed the `json:URI` bit. So this looks good to me. Could you give this PR a better title? I know it has a few different changes in it, but maybe the title can be the most important change which is to `SELECT name`.

Done.

**Pull request merged**